synapse/handlers/room_summary.py

```diff
@@ -701,13 +700,6 @@ async def _build_room_entry(self, room_id: str, for_federation: bool) -> JsonDic
         # there should always be an entry
         assert stats is not None, "unable to retrieve stats for %s" % (room_id,)
 
-        current_state_ids = await self._storage_controllers.state.get_current_state_ids(
-            room_id
-        )
```

Comment:
Another approach might have been to use a StateFilter to only fetch the create event.

Reply:
Sure, but you're still doing additional queries you don't need!